### PR TITLE
Make large cactus seedling dies under liquid

### PR DESCRIPTION
**Pull request by bhree**
Players will not try planting them underwater which will prevent seedling from growing and keep ticking.

### CODE REVIEW

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -1323,13 +1323,16 @@ minetest.register_node("default:large_cactus_seedling", {
...
+			{x = pos.x, y = pos.y + 1, z = pos.z})
 		if not node_under then
 			-- Node under not yet loaded, try later
 			minetest.get_node_timer(pos):start(300)
 			return
 		end
 
-		if minetest.get_item_group(node_under.name, "sand") == 0 then
+		if minetest.get_item_group(node_under.name, "sand") == 0
+				or minetest.get_item_group(node_above.name, "liquid") >= 1 then
```
Comment (sfan5) [inline mods/default/nodes.lua]: `node_above` might be nil here

**Comment by paramat** [comment]
However, a player could also plant one in a cave or a dark place, which would also keep the timer ticking, so i can't see any point in checking just one type of weird planting location. It's highly unlikely a player will plant a cactus seedling underwater.
We would then need to do this for all plants too. It's unnecessary and unjustified complexity.
:-1: 
However thank you for your effort and involvement, it's appreciated.

**Comment by Ezhh** [comment]
It doesn't make sense to have such a check only for one specific type of plant.

**Comment by paramat** [comment]
However, to be clear, i oppose underwater checks being added for anything.

**Comment by Ezhh** [comment]
> However, to be clear, i oppose underwater checks being added for anything.

Same.

**Comment by benrob0329** [comment]
Alternative proposal: Make them floodable.

**Comment by paramat** [comment]
Making plants floodable has been suggested and rejected before, because water griefing or accidental spills becomes so much more destructive.

**Pull request closed without merging**